docs/ibc/overview.md

```diff
@@ -49,6 +58,60 @@ The supported IBC clients are:
 testing, simulation, and relaying packets to modules on the same application.
 
+### IBC Client Heights
+
+IBC Client Heights are represented by the struct:
+
+```go
+type Height struct {
+   RevisionNumber uint64
+   RevisionHeight uint64
+}
+```
+
+The `RevisionNumber` represents the revision of the chain that the height is representing.
+A revision typically represents a continuous, monotonically increasing range of block-heights.
+The `RevisionHeight` represents the height of the chain within the given revision.
+
+On any reset of the `RevisionHeight`, for example, when hard-forking a Tendermint chain,
+the `RevisionNumber` will get incremented. This allows IBC clients to distinguish between a
+block-height `n` of a previous revision of the chain (at revision `p`) and block-height `n` of the current
+revision of the chain (at revision `e`).
+
+`Heights` that share the same revision number can be compared by simply comparing their respective `RevisionHeights`.
+Heights that do not share the same revision number will only be compared using their respective `RevisionNumbers`.
+Thus a height `h` with revision number `e+1` will always be greater than a height `g` with revision number `e`,
+**REGARDLESS** of the difference in revision heights.
+
+Ex:
+
+```go
+Height{RevisionNumber: 3, RevisionHeight: 0} > Height{RevisionNumber: 2, RevisionHeight: 100000000000}
+```
+
+When a Tendermint chain is running a particular revision, relayers can simply submit headers and proofs with the revision number
+given by the chain's chainID, and the revision height given by the Tendermint block height. When a chain updates using a hard-fork 
+and resets its block-height, it is responsible for updating its chain-id to increment the revision number.
+IBC Tendermint clients then verifies the revision number against their `ChainId` and treat the `RevisionHeight` as the Tendermint block-height.
+
+Tendermint chains wishing to use revisions to maintain persistent IBC connections even across height-resetting upgrades must format their chain-ids
+in the following manner: `{chainID}-{revision_number}`. On any height-resetting upgrade, the chainID **MUST** be updated with a higher revision number
+than the previous value.
+
+Ex:
+
+- Before upgrade ChainID: `gaiamainnet-3`
+- After upgrade ChainID: `gaiamainnet-4`
+
+Clients that do not require revisions, such as the solo-machine client, simply hardcode `0` into the revision number whenever they
+need to return an IBC height when implementing IBC interfaces and use the `RevisionHeight` exclusively.
+
+Other client-types may implement their own logic to verify the IBC Heights that relayers provide in their `Update`, `Misbehavior`, and
+`Verify` functions respectively.
+
+The IBC interfaces expect an `ibcexported.Height` interface, however all clients should use the concrete implementation provided in
+`02-client/types` and reproduced above.
```

Comment:
What does it mean `and reproduced above`?

Reply:
I think the concrete `Height` type is defined in the docs (reproduced). I didn't write these docs so I'm not 100% sure